File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */

import java.util.List;
import java.util.ArrayList;
import java.util.Queue;

class Solution {
    public List<List<Integer>> levelOrder(TreeNode root) {
        List<List<Integer>> result = new ArrayList<List<Integer>>();
        Queue<TreeNode> q = new LinkedList<TreeNode>();
        q.add(root);
        while(q.size()>0){
            List<Integer> temp = new ArrayList<Integer>();
            int l = q.size();
            for (int i = 0; i < l; i++){
                TreeNode node = q.peek();
                q.remove();
                if(node != null){
                    temp.add(node.val);
                    q.add(node.left);
                    q.add(node.right);
                }
            }
            if(temp.size()>0){
                result.add(temp);
            }
        }
        return result;
    }
}

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
#include<queue>
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> result;
        queue<TreeNode*> q;
        q.push(root);
        while(q.size()>0){
            vector<int> temp;
            int l = q.size();
            for (int i = 0; i < l; i++){
                TreeNode* node = q.front();
                q.pop();
                if(node != nullptr){
                    temp.push_back(node->val);
                    q.push(node->left);
                    q.push(node->right);
                }
            }
            if(temp.size()>0){
                result.push_back(temp);
            }
        }
        return result;
    }
};